### PR TITLE
Respect per RPC creds even during Dial()

### PR DESCRIPTION
**Pull request by gkousik**
Also added an option "use_external_auth_token" that when set to true, would mandate the usae of per RPC creds option starting with the initial Dial to setup the connection

Test: Also implemented the corresponding re-client side of things and ensured that reproxy starts up

### CODE REVIEW

File: go/pkg/flags/flags.go [hunk 1]
```diff
@@ -33,6 +33,8 @@ var (
 	UseGCECredentials = flag.Bool("use_gce_credentials", false, "If true (and --use_application_default_credentials is false), use the default GCE credentials to authenticate with remote execution.")
 	// UseRPCCredentials can be set to false to disable all per-RPC credentials.
 	UseRPCCredentials = flag.Bool("use_rpc_credentials", true, "If false, no per-RPC credentials will be used (disables --credential_file, --use_application_default_credentials, and --use_gce_credentials.")
+	// UseExternalAuthToken specifies whether to use an externally provided auth token, given via PerRPCCreds dial option, should be used.
+	UseExternalAuthToken = flag.Bool("use_external_auth_token", false, "If true, uses a blah blah blah.")
```
Comment (ramymedhat) [inline go/pkg/flags/flags.go:37]: Fix comment?

File: go/pkg/flags/flags.go [hunk 2]
```diff
@@ -93,6 +95,14 @@ func NewClientFromFlags(ctx context.Context, opts ...client.Opt) (*client.Client
 		}
 		opts = append(opts, client.RPCTimeouts(timeouts))
 	}
+	var perRPCCreds *client.PerRPCCreds
+	for _, opt := range opts {
+		switch opt.(type) {
```
Comment (ramymedhat) [inline go/pkg/flags/flags.go:100]: Just an if would suffice?

**Review by ramymedhat — APPROVED** [review]

File: go/pkg/flags/flags.go (hunk 2 above)
Comment (gkousik) [reply go/pkg/flags/flags.go:100]: ah actually its not possible to use `.(type)` outside a switch case - https://stackoverflow.com/questions/34820469/can-i-compare-variable-types-with-type-in-golang.

File: go/pkg/flags/flags.go (hunk 1 above)
Comment (gkousik) [reply go/pkg/flags/flags.go]: Ah embarassing :D done!

**Review by ramymedhat — APPROVED** [review]

**Pull request merged**